Reflection/Combat.simba

```diff
@@ -0,0 +1,280 @@
+{$include_once Internal/Reflection.simba}
+{$include_once Constants.simba}
+{$include_once GameTab.simba}
+{$include_once Mouse.simba}
+{$include_once Timing.simba}
+
+type
+  TAttackSpell = (
+    WIND_STRIKE = 1,
+    WATER_STRIKE,
+    EARTH_STRIKE,
+    FIRE_STRIKE,
+    WIND_BOLT,
+    WATER_BOLT,
+    EARTH_BOLT,
+    FIRE_BOLT,
+    WIND_BLAST,
+    WATER_BLAST,
+    EARTH_BLAST,
+    FIRE_BLAST,
+    WIND_WAVE,
+    WATER_WAVE,
+    EARTH_WAVE,
+    FIRE_WAVE);
+
+(*
+COMBAT
+======
+
+  Functions that interact with the Combat tab.
+  Examples:
+    - Get all available attack styles
+    - Change attack styles
+    - Use special attack
+    - Enable Auto Retaliate
+*)
+
+{*
+R_SelectAttackStyle
+~~~~~~~~~~~~~~~~~~~~~~
+Sets the attack style to the specified attack style.
+  Options:
+    Smash
+    Block
+    Impale
+    Spike
+    ...
+
+Example:
+    R_SelectAttackStyle("Pound");
+*}
+Function R_SelectAttackStyle(Style: String; ReturnToTab: Boolean = False): Boolean;
+var
+  Widget, CheckWidget: RSWidget;
+  WeaponStyles: Array of RWidget;
+  I, PreviousTab: Integer;
+begin
+  WeaponStyles := [R_COMBAT_STYLE_ONE, R_COMBAT_STYLE_TWO, R_COMBAT_STYLE_THREE,R_COMBAT_STYLE_FOUR];
+
+  for I := 0 to High(WeaponStyles) do
+  begin
+    Widget := RSWidget.Get(WeaponStyles[I].Group, WeaponStyles[I].Child + 3);
```

Comment:
I've used it in this fashion to prevent from having to take note of too many constants. Especially if you are dealing with a relatively obscure widget that will most likely not be used anywhere else. I have no problem with implementing it as a constant if that were thought to be better.

Reply:
I don't have too much of a problem with the offset as is, just curious how it is determined, really.